spring-cloud-aws-autoconfigure/src/main/java/io/awspring/cloud/autoconfigure/config/appconfig/AppConfigDataLocationResolver.java

```diff
@@ -0,0 +1,118 @@
+import io.awspring.cloud.autoconfigure.core.AwsProperties;
+import io.awspring.cloud.autoconfigure.core.CredentialsProperties;
+import io.awspring.cloud.autoconfigure.core.RegionProperties;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+import org.apache.commons.logging.Log;
+import org.springframework.boot.BootstrapContext;
+import org.springframework.boot.context.config.ConfigDataLocation;
+import org.springframework.boot.context.config.ConfigDataLocationNotFoundException;
+import org.springframework.boot.context.config.ConfigDataLocationResolverContext;
+import org.springframework.boot.context.config.Profiles;
+import org.springframework.boot.context.properties.bind.Bindable;
+import org.springframework.boot.context.properties.bind.Binder;
+import software.amazon.awssdk.services.appconfigdata.AppConfigDataClient;
+import software.amazon.awssdk.services.appconfigdata.AppConfigDataClientBuilder;
+
+public class AppConfigDataLocationResolver extends AbstractAwsConfigDataLocationResolver<AppConfigDataResource> {
+
+	public static final String PREFIX = "aws-appconfig:";
+
+	private final Log log;
+
+	public AppConfigDataLocationResolver(Log log) {
+		this.log = log;
+	}
+
+	@Override
+	protected String getPrefix() {
+		return PREFIX;
+	}
+
+	@Override
+	public List<AppConfigDataResource> resolveProfileSpecific(ConfigDataLocationResolverContext resolverContext,
+			ConfigDataLocation location, Profiles profiles) throws ConfigDataLocationNotFoundException {
+		registerBean(resolverContext, AwsProperties.class, loadAwsProperties(resolverContext.getBinder()));
+		registerBean(resolverContext, AppConfigProperties.class, loadProperties(resolverContext.getBinder()));
+		registerBean(resolverContext, CredentialsProperties.class,
+				loadCredentialsProperties(resolverContext.getBinder()));
+		registerBean(resolverContext, RegionProperties.class, loadRegionProperties(resolverContext.getBinder()));
+
+		registerAndPromoteBean(resolverContext, AppConfigDataClient.class, this::createAppConfigDataClient);
+
+		AppConfigPropertySources sources = new AppConfigPropertySources();
+
+		List<String> contexts = getCustomContexts(location.getNonPrefixedValue(PREFIX));
+
+		List<AppConfigDataResource> locations = new ArrayList<>();
+		contexts.forEach(propertySourceContext -> locations
+				.add(new AppConfigDataResource(splitContext(propertySourceContext), location.isOptional(), sources)));
+
+		if (!location.isOptional() && locations.isEmpty()) {
+			throw new AppConfigKeysMissingException(
+					"No AppConfig keys provided in `spring.config.import=aws-appconfig:` configuration.");
+		}
+		return locations;
+	}
+
+	public static RequestContext splitContext(String context) {
+		RequestContext.Builder builder = RequestContext.builder();
+		AtomicReference<Integer> i = new AtomicReference<>(0);
+		builder.context(context);
+		Arrays.stream(context.split("___")).forEach(split -> {
```

Comment:
Should the prefix that is splitting values be adjustable or changed to something else? 
I think even a simple `/` could do but I am afraid there will be cases where someone has `/` in ConfigurationProfileName or ApplicationName

Reply:
I would go with the simple delimiter `/` and wait for feedback on the next milestone. Given you can have any character in the names. Also, for better debugging trace for loop would be enough